mesonpy/__init__.py

```diff
@@ -558,7 +568,8 @@ def build(self, directory: Path) -> pathlib.Path:
 
                 # install bundled libraries
                 for destination, origin in self._wheel_files['mesonpy-libs']:
-                    assert platform.system() == 'Linux', 'Bundling libraries in wheel is currently only supported in POSIX!'
+                    assert platform.system() == 'Linux' or platform.system() == 'Darwin', \
+                        'Bundling libraries in wheel is currently only supported in POSIX!'
```

Comment:
Sorry for the intrusion, but I also looked into this code and wanted to ask a question.

Why is the assert placed in line 571; this is remote from the location where the code actually fails to handle the platform. It makes the code a bit harder to understand than necessary. I would do an `elif` in line 529 and then a final `else`  branch and put the assert there.

Reply:
This makes a lot of sense, I've moved the failure to the point you suggested. Note that is is still an assert, should this become an exception?

Reply:
Good question. I agree that an exception is better than an `assert`. Asserts are for things that are not supposed to happen, and may be stripped out on install.

Reply:
A `NotImplementedError` would probably be appropriate. I'll update the code accordingly.

Reply:
I changed this to a `NotImplementedError`